lib/widgets/sign_up_widget.dart

```diff
@@ -0,0 +1,174 @@
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+import '../configuration/app_text.dart';
+import '../screens/chat_rooms_screen.dart';
+import '../configuration/app_path.dart';
+import '../configuration/app_colors.dart';
+import '../services/auth.dart';
+
+class SignUp extends StatefulWidget {
+  @override
+  _SignUpState createState() => _SignUpState();
+}
+
+class _SignUpState extends State<SignUp> {
+  bool isLoading = false;
+
+  AuthMethods authMethods = new AuthMethods();
+
+  final formKey = GlobalKey<FormState>();
+  TextEditingController userNameTextEditingController =
+      new TextEditingController();
+  TextEditingController emailTextEditingController =
+      new TextEditingController();
+  TextEditingController passwordTextEditingController =
+      new TextEditingController();
+
+  signMeUp() {
+    if (formKey.currentState.validate()) {
+      setState(() {
+        isLoading = true;
+      });
+      authMethods
+          .singUpwithEmailAndPassword(emailTextEditingController.text,
+              passwordTextEditingController.text)
+          .then((value) {
+        //print("${value.uid}");
+
+        Navigator.pushReplacement(
+            context, MaterialPageRoute(builder: (context) => ChatRoom()));
+      });
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: Image.asset(
+          AppPath.logoAsset,
+          height: 40,
+          color: AppColors.primaryIconColor,
+        ),
+      ),
+      body: isLoading
+          ? Container(
+              child: CircularProgressIndicator(),
+              alignment: AlignmentDirectional.center)
+          : SingleChildScrollView(
+              child: Container(
+                height: MediaQuery.of(context).size.height - 60,
+                alignment: Alignment.bottomCenter,
+                child: Container(
+                  padding: EdgeInsets.symmetric(horizontal: 25),
+                  child: Column(
+                    mainAxisSize: MainAxisSize.min,
+                    children: [
+                      Form(
+                        key: formKey,
+                        child: Column(
+                          children: [
+                            TextFormField(
```

Comment:
text form field could be moved to a separate build method with required parameters, example 
`_buildTextFormField({Function validatorFunction, TextEditingController controller, String hintText, bool obscureText = false,})`, because everything else is similar you will avoid doubling the code and this will reduce amount of code in your app